testing/go/set_test.go

```diff
@@ -0,0 +1,7352 @@
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    "SHOW application_name",
+				Expected: []sql.Row{{"psql"}},
+			},
+			{
+				Query:    "SET application_name TO 'postgresql'",
+				Expected: []sql.Row{{}},
+			},
+			{
+				Query:    "SHOW application_name",
+				Expected: []sql.Row{{"postgresql"}},
+			},
+			{
+				Query:    "SET application_name TO DEFAULT",
+				Expected: []sql.Row{{}},
+			},
+			{
+				Query:    "SHOW application_name",
+				Expected: []sql.Row{{"psql"}},
+			},
+		},
+	},
+	{
+		Name:        "set 'archive_cleanup_command' configuration variable",
+		SetUpScript: []string{},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    "SHOW archive_cleanup_command",
+				Expected: []sql.Row{{""}},
+			},
+			{
+				Query:       "SET archive_cleanup_command TO ''",
+				ExpectedErr: true,
+			},
+		},
+	},
+	{
+		Name:        "set 'archive_command' configuration variable",
+		SetUpScript: []string{},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    "SHOW archive_command",
+				Expected: []sql.Row{{"(disabled)"}},
+			},
+			{
+				Query:       "SET archive_command TO ''",
+				ExpectedErr: true,
+			},
+		},
+	},
+	{
+		Name:        "set 'archive_library' configuration variable",
+		SetUpScript: []string{},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    "SHOW archive_library",
+				Expected: []sql.Row{{""}},
+			},
+			{
+				Query:       "SET archive_library TO ''",
+				ExpectedErr: true,
+			},
+		},
+	},
+	{
+		Name:        "set 'archive_mode' configuration variable",
+		SetUpScript: []string{},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    "SHOW archive_mode",
+				Expected: []sql.Row{{"off"}},
+			},
+			{
+				Query:       "SET archive_mode TO 'off'",
+				ExpectedErr: true,
+			},
+		},
+	},
+	{
+		Name:        "set 'archive_timeout' configuration variable",
+		SetUpScript: []string{},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    "SHOW archive_timeout",
+				Expected: []sql.Row{{int64(0)}},
+			},
+			{
+				Query:       "SET archive_timeout TO '0'",
+				ExpectedErr: true,
+			},
+		},
+	},
+	{
+		Name:        "set 'array_nulls' configuration variable",
+		SetUpScript: []string{},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    "SHOW array_nulls",
+				Expected: []sql.Row{{int8(1)}},
+			},
+			{
+				Query:    "SET array_nulls TO 'off'",
+				Expected: []sql.Row{{}},
+			},
+			{
+				Query:    "SHOW array_nulls",
+				Expected: []sql.Row{{int8(0)}},
+			},
+			{
+				Query:    "SET array_nulls TO DEFAULT",
+				Expected: []sql.Row{{}},
+			},
+			{
+				Query:    "SHOW array_nulls",
+				Expected: []sql.Row{{int8(1)}},
+			},
+		},
+	},
+	{
+		Name:        "set 'authentication_timeout' configuration variable",
+		SetUpScript: []string{},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    "SHOW authentication_timeout",
+				Expected: []sql.Row{{int64(60)}},
+			},
+			{
+				Query:       "SET authentication_timeout TO '120'",
```

Comment:
Are these tests just showing that we do not allow setting the value yet? It seems like it shouldn't error, `'120'` seems like a valid value. If that's the case, then these should be skipped tests for the correct result, rather than reinforcing that the error is the correct result. I see this pattern on a lot of tests here, so I'm not sure.

Reply:
This is of context `sighup`, which means these variables cannot be set using the SET statement. According to the docs, it says it can be updated using a config file and sending SIGHUP signal to `postmaster`, which I think is something we don't support currently. So, the current implementation only allows SET variable to variables of context `user` or `superuser`, but it needs more validation as we support `superuser` and privilege checks.